src/librustc_codegen_llvm/common.rs

```diff
@@ -260,9 +260,17 @@ impl ConstMethods<'tcx> for CodegenCx<'ll, 'tcx> {
                     Some(GlobalAlloc::Memory(alloc)) => {
                         let init = const_alloc_to_llvm(self, alloc);
                         if alloc.mutability == Mutability::Mut {
-                            self.static_addr_of_mut(init, alloc.align, None)
+                            self.static_addr_of_mut(
+                                init,
+                                alloc.align,
+                                Some(format!("{:?}", ptr.alloc_id).as_str()),
+                            )
                         } else {
-                            self.static_addr_of(init, alloc.align, None)
+                            self.static_addr_of(
+                                init,
+                                alloc.align,
+                                Some(format!("{:?}", ptr.alloc_id).as_str()),
```

Comment:
In the back&forth it looks like you lost the deduplication here, which my primary concern.
Having two copies of this line makes it harder to discuss and/or change the logic.

Reply:
963592c3f26ec32d33797f356f252dc558030269 returns to the original approach with fewer_names check and deduplication